src/openrct2/actions/GameActionResult.h

```diff
@@ -70,7 +70,12 @@ namespace GameActions
         CoordsXYZ Position = { LOCATION_NULL, LOCATION_NULL, LOCATION_NULL };
         money32 Cost = 0;
         ExpenditureType Expenditure = ExpenditureType::Count;
-        std::shared_ptr<void> ResultData;
+
+        // Any_cast throws a bad_any_cast exception on Android
+        // To avoid this in the Android release, a shared void pointer is used to store the result data.
+        // Other platforms still use Any as this provides type checks
+        std::shared_ptr<void> ResultDataPtr;
+        std::any ResultData;
```

Comment:
```suggestion
#ifdef __ANDROID__
        // Any_cast throws a bad_any_cast exception on Android
        // To avoid this in the Android release, a shared void pointer is used to store the result data.
        std::shared_ptr<void> ResultData;
#else
        // Other platforms still use Any as this provides type checks
        std::any ResultData;
#endif
```

I think we should also branch here and keep just one member variable.